lib/msf/core/exploit/remote/http/nifi/dbconnectionpool.rb

```diff
@@ -0,0 +1,77 @@
+# -*- coding: binary -*-
+
+module Msf::Exploit::Remote::HTTP::Nifi::Dbconnectionpool
+  # Stop DB Connection Pool
+  #
+  # @param token [String] The bearer token from a valid login
+  # @param db_con_pool [String] UUID of the DBConnectionPool
+  def stop_dbconnectionpool(token, db_con_pool)
+    body = {
+      'disconnectedNodeAcknowledged' => false,
+      'state' => 'DISABLED',
+      'uiOnly' => true,
+      'revision' => {
+        'clientId' => 'x',
+        'version' => 0
+      }
+    }
+    opts = {
+      'method' => 'PUT',
+      'uri' => normalize_uri(target_uri.path, 'nifi-api', 'controller-services', db_con_pool, 'run-status'),
+      'ctype' => 'application/json',
+      'data' => body.to_json
+    }
+    opts['headers'] = { 'Authorization' => "Bearer #{token}" } if token
+    response = send_request_cgi(opts)
+    check_response("PUTting disable status for db connection pool #{db_con_pool}", response, 200)
+  end
+
+  # Delete DB Connection Pool
+  #
+  # @param token [String] The bearer token from a valid login
+  # @param db_con_pool [String] UUID of the DBConnectionPool
+  def delete_dbconnectionpool(token, db_con_pool)
+    version = 0
+    opts = {
+      'method' => 'DELETE',
+      'uri' => normalize_uri(target_uri.path, 'nifi-api', 'controller-services', db_con_pool),
+      'vars_get' => {
+        'version' => version
+      }
+    }
+    opts['headers'] = { 'Authorization' => "Bearer #{token}" } if token
+    response = send_request_cgi(opts)
+    while response.code == 400 && response.body.include?('is not the most up-to-date revision') && version <= 20
```

Comment:
If we _do_ hit 20, and we haven't gotten the most up-to-date revision - should we raise an error? As something's probably gone very wrong in that scenario 😄

Reply:
Is there anything that will indicate a positive success here that we can cross-check?

Also are there any language compatibility issues we need to consider here, i.e. can this server respond in french etc

Reply:
Would this break if the server returned a 404 or similar?